src/Adapter/MSTest.TestAdapter/Extensions/MethodInfoExtensions.cs

```diff
@@ -162,18 +162,36 @@ internal static void InvokeAsSynchronousTask(this MethodInfo methodInfo, object?
         {
             int methodParametersLengthOrZero = methodParameters?.Length ?? 0;
             int argumentsLengthOrZero = arguments?.Length ?? 0;
-            if (methodParametersLengthOrZero != argumentsLengthOrZero)
+
+#if WINDOWS_UWP
+            // There is a bug with UWP in release mode where the arguments are wrapped in an object[], so we need to unwrap it.
+            // See https://github.com/microsoft/testfx/issues/3071
+            if (argumentsLengthOrZero == 1
+                && argumentsLengthOrZero < methodParametersLengthOrZero
+                && arguments![0] is object[] args)
```

Comment:
We could be even more restrictive and only unwrap if `args.Length == methodParametersLengthOrZero`, WDYT?